astropy/coordinates/tests/test_representation.py

```diff
@@ -1046,6 +1049,16 @@ class LogDRepresentation(BaseRepresentation):
                                         ('lat', Latitude),
                                         ('logr', u.Dex)])
 
+    # now try passing in a differential
+    diff = CartesianDifferential(d_x=1 * u.km/u.s,
```

Comment:
If you just use the automatic constructor, `DifferentialMeta` will assume that all arguments are `Quantity`, and hence `u.Dex` instances will fail. But those are actually not expected: the unit you should use is `u.dex`, since a difference in `logd` should be unitless:
```
In [10]: u.Dex(1.*u.kpc) - u.Dex(0.5*u.kpc)
Out[10]: <Dex 0.3010299956639812 dex>
```
Of course, for transformations to work, you also need to define `unit_vectors` and `scale_factors` in the class...

If it helps, I can make a PR against current master that tests this.